File: frontend/src/types/horse.ts
export interface HorseDocument {
  type: "Health Certificate" | "Registration Paper" | "Vaccination Record" | "Pedigree" | "Insurance";
  number: string;
  issuedBy: string;
  issuedDate: string;
  expiryDate?: string;
  fileUrl?: string;
}

export interface Horse {
  id: string;
  name: string;
  breed: string;
  age: number;
  weight: number;
  ownerId: string;
  healthCertExpiry: string;
  status: "Eligible" | "Ineligible" | "Suspended";
  sire?: string;          
  dam?: string;         
  color?: string;
  microchipId?: string;
  trainer?: string;
  bio?: string;
  documents: HorseDocument[];
}

File: HRTMS/HRTMS/Models/Horse.cs
using System;
using System.Collections.Generic;

namespace HRTMS.Models;

public partial class Horse
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Breed { get; set; }

    public int? Age { get; set; }

    public decimal? Weight { get; set; }

    public int? OwnerId { get; set; }

    public DateOnly? HealthCertExpiry { get; set; }

    public string? Status { get; set; }

    public string? Color { get; set; }

    public string? Trainer { get; set; }

    public string? Bio { get; set; }

    public virtual Owner? Owner { get; set; }

    public virtual ICollection<RaceResult> RaceResults { get; set; } = new List<RaceResult>();

    public virtual ICollection<Registration> Registrations { get; set; } = new List<Registration>();

    public virtual ICollection<Violation> Violations { get; set; } = new List<Violation>();
}
